File: miraculous-backend/src/main/java/com/example/miraculousbackend/dto/PlayableTrackDto.java
package com.example.miraculousbackend.dto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class PlayableTrackDto {

    private Long id;
    private String title;
    private Long artistId;
    private String artistLogin;
    private Long albumId;
    private String albumTitle;
    private Double duration;
    private Long numberOfLikes;
    private String coverPictureUrlName;
    private String audioUrlName;
    private Boolean isFree;
    private Double price;
    private List<String> genres;
    private String dateOfRelease;
    private boolean isLikedByCurrentVisitor;
    private boolean isInCartOfCurrentVisitor;

}

File: miraculous-frontend/miraculous-ui/src/dto/playable-track-dto/playable-track-dto.ts
export class PlayableTrackDto {

  id: number;
  title: string;
  artistId: number;
  artistLogin: string;
  albumId: number;
  albumTitle: string;
  duration: number;
  numberOfLikes: number;
  coverPictureUrlName: string;
  audioUrlName: string;
  isFree: boolean;
  price: number;
  genres: string[];
  dateOfRelease: string;
  inCartOfCurrentVisitor: boolean;
  likedByCurrentVisitor: boolean;


  constructor(id?: number, title?: string, artistId?: number,
              artistLogin?: string, albumId?: number, albumTitle?: string,
              duration?: number, numberOfLikes?: number, coverPictureUrl?: string, audioUrl?: string,
              isFree?: boolean, price?: number, genres?: string[], dateOfRelease?: string,
              isInCurrentVisitorCart?: boolean, isLikedByCurrentVisitor?: boolean) {
    this.id = id;
    this.title = title;
    this.artistId = artistId;
    this.artistLogin = artistLogin;
    this.albumId = albumId;
    this.albumTitle = albumTitle;
    this.duration = duration;
    this.numberOfLikes = numberOfLikes;
    this.coverPictureUrlName = coverPictureUrl;
    this.audioUrlName = audioUrl;
    this.isFree = isFree;
    this.price = price;
    this.genres = genres;
    this.dateOfRelease = dateOfRelease;
    this.inCartOfCurrentVisitor = isInCurrentVisitorCart;
    this.likedByCurrentVisitor = isLikedByCurrentVisitor
  }
}

